Subject: video: Prefer "bochs" when it's supported.
From: Fabiano Fidêncio fidencio@redhat.com Thu Oct 3 10:50:34 2019 +0200
Date: Fri Oct 4 11:17:10 2019 -0400:
Git: 8f4c53ea960459516794ba533060a176cc26f121

Preferring "bochs" display device is the way to go when dealing with a
Linux guest using UEFI and that's quite well described here:
https://www.kraxel.org/blog/2019/09/display-devices-in-qemu/

https://bugzilla.redhat.com/show_bug.cgi?id=1753644

Signed-off-by: Fabiano Fidêncio <fidencio@redhat.com>

diff --git a/virtinst/devices/video.py b/virtinst/devices/video.py
index 3ebc561f..fcca91b6 100644
--- a/virtinst/devices/video.py
+++ b/virtinst/devices/video.py
@@ -39,6 +39,9 @@ class DeviceVideo(Device):
             if guest.has_gl():
                 return "virtio"
             return "qxl"
+        if (guest.is_uefi() and
+            guest.lookup_domcaps().supports_video_bochs()):
+            return "bochs"
         if guest.conn.is_qemu():
             return "qxl"
         return "vga"
